### ISSUE TITLE
If you set the inputtime to 8:30 the control makes it 9:30

### ISSUE THREAD

**Issue by dickverweij** [rajeshwarpatlolla/ionic-timepicker#86]
This is because you use toFixed(0) to round the value for the hours.
You need to use Math.floor..

![image](https://cloud.githubusercontent.com/assets/8133838/15925689/7d260ee6-2e38-11e6-9bd2-2e2f0d23927a.png)

**Comment by DharmendraVaishnav91** [comment]
+1  I am facing the same issue.

**Comment by lucasdoamaral** [comment]
I need to write this junky piece of code, to fix this in runtime.

![image](https://cloud.githubusercontent.com/assets/9009346/16569824/aaa7ada6-4211-11e6-8bea-4b331112b8d3.png)

Even with the commit made by @koiyu the problem continues to happen. Any news about that?

**Comment by keinajar** [comment]
@lucasdoamaral I don't understand your problem without more context. 

If your `currentTime` represents `inputTime` option, then remember: `inputTime` is not a `Date` object. 

If you wish to use `Date` objects (I do, at least), then you could use the following options:

```
$scope.data.created = new Date();

$scope.changeTime = function() {
            ionicTimePicker.openTimePicker({
                callback: function(val) {
                    var hours = Math.floor(val / 60 / 60),
                        minutes = val / 60 % 60;
                    $scope.data.created.setHours(hours, minutes);
                },
                inputTime: ((($scope.data.created.getHours() * 60 * 60) + (($scope.data.created.getMinutes() * 60)))),
                format: 24
            });
        };
```

**Comment by lync0056** [comment]
Hi @koiyu ,

What they are saying is correct. I am experiencing the same issue. With your code, if I open, the current time in midday (12.03pm). The input on 12 hour clock time will show as  00 : 03 pm. I think before that I did it at 11.55am and it showed as 12.55am.

Very useful Directive. Thank you.

Cameron

**Comment by vikramaditya234** [comment]
+1

**Comment by vikramaditya234** [comment]
+1 still the issue, please regenerate ionic-timepicker.bundle.min.js
[this is the fix](https://github.com/rajeshwarpatlolla/ionic-timepicker/issues/86)

**Comment by osirisr** [comment]
Had the same problem (toFixed was rounding the time). @koiyu fix worked. Really wish I checked here first instead of spending 3 hours finding the bug myself.

**Comment by brunodeprez** [comment]
Hi,

could you push the new version of this component to bower, including your fix of this issue ?

Thanks

**Comment by rajeshwarpatlolla** [comment]
Fixed in [v0.6.0](https://github.com/rajeshwarpatlolla/ionic-timepicker/releases/tag/v0.6.0)
